interface.go

```diff
@@ -55,12 +55,54 @@ type Interface interface {
 	Stats() (*Stats, error)
 }
 
+type Datastore interface {
+	// StartBatch starts a new batch operation that is specific to the datastore implementation
+	StartBatch() interface{}
+
+	// CommitBatch commits a batch that is specific to the datastore implementation
+	CommitBatch(batch interface{}) error
+
+	// CloseBatch closes a batch that is specific to the datastore implementation
+	CloseBatch(batch interface{}) error
+
+	// GetKeyValues returns value keys associated with the multihash
+	GetKeyValues(multihash []byte) ([][]byte, bool, error)
+
+	// GetValue returns a value associated with the value key
+	GetValue(valKey []byte) (*Value, error)
+
+	// PutValue puts a value into the datastore and associates it with the value key using provided batch
+	PutValue(value Value, batch interface{}) ([]byte, error)
+
+	// PutKeyValue puts a new mapping from multihash to the value key using provided batch
+	PutKeyValue(multihash []byte, valKey []byte, batch interface{}) error
+
+	// RemoveKeyValue removes a value key associated with the multihash
+	RemoveKeyValue(mh []byte, valKey []byte, batch interface{}) error
+
+	// RemoveValue removes a value from the datastore
+	RemoveValue(value *Value) error
+
+	// ValueKey creates a value key for the given value. Different datastpre might have different ways of composing that.
+	ValueKey(value *Value) ([]byte, error)
+
+	Size() (int64, error)
+
+	Flush() error
+
+	Close() error
+
+	Iter() (Iterator, error)
+
+	Stats() (*Stats, error)
+}
+
 // Iterator iterates multihashes and values in the value store. Any write
 // operation invalidates the iterator.
 type Iterator interface {
 	// Next returns the next multihash and the value it indexer. Returns io.EOF
 	// when finished iterating.
-	Next() (multihash.Multihash, []Value, error)
+	Next() ([]byte, [][]byte, error)
```

Comment:
I thought there will be a multihash that signifies "doublehashed" entires. In which case I am curious why the type is replaced with []byte?

Reply:
True, that can be changed back to multihash